battery-upower/battery-upower

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# Copyright 2020 Sebastian Wiesner <sebastian@swsnr.de>
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+
+import os
+import sys
+import json
+from enum import IntEnum
+from gi.repository import Gio, GLib
+
+
+class DeviceType(IntEnum):
+    """See https://upower.freedesktop.org/docs/Device.html#Device:Type"""
+    BATTERY = 2
+
+
+class DeviceState(IntEnum):
+    """See https://upower.freedesktop.org/docs/Device.html#Device:State"""
+    CHARGING = 1
+    FULLY_CHARGED = 4
+
+
+class DeviceWarningLevel(IntEnum):
+    """See https://upower.freedesktop.org/docs/Device.html#Device:WarningLevel"""
+    LOW = 3
+    CRITICAL = 4
+    ACTION = 5
+
+
+def find_battery(upower):
+    for device_path in upower.EnumerateDevices():
+        device = Gio.DBusProxy.new_sync(
+            upower.get_connection(), Gio.DBusProxyFlags.NONE,
+            None, upower.get_name(), device_path,
+            'org.freedesktop.UPower.Device')
+        if device.get_cached_property('Type').unpack() == DeviceType.BATTERY:
+            return device
+    else:
```

Comment:
I think you don't need a `for/else` loop here because on line 49 you have a `return` statement that would continue the execution of the code in the `main()` function. The `for/else` loop is useful for when you have a `break` statement in your `for` loop because the execution of the code would continue in the same function where the `for` loop is declared. This is not the case.

I'd suggest changing the `for/else` for a `for` loop and returning at the end, like this:

```python3
def find_battery(upower):
    for device_path in upower.EnumerateDevices():
        device = Gio.DBusProxy.new_sync(
            upower.get_connection(), Gio.DBusProxyFlags.NONE,
            None, upower.get_name(), device_path,
            'org.freedesktop.UPower.Device')
        if device.get_cached_property('Type').unpack() == DeviceType.BATTERY:
            return device
    return None
```

What do you think?